rate_limit.py

```diff
@@ -0,0 +1,98 @@
+"""
+This module contains functions for rate limiting requests.
+"""
+
+import datetime
+import signal
+import sys
+from uuid import uuid4
+
+from apscheduler.schedulers import background
+import gradio as gr
+
+
+class InvalidTokenException(Exception):
+  pass
+
+
+class UserRateLimitException(Exception):
+  pass
+
+
+class SystemRateLimitException(Exception):
+  pass
+
+
+class RateLimiter:
+
+  def __init__(self, daily_limit=10000):
+    self.requests = {}
```

Comment:
It will be good if you add comments or type annotations.
It's hard to know what this dict is supposed to look like.

Reply:
I agree